### ISSUE TITLE
setup.py setup-db triggers file not found on first run of pgadmin4

### ISSUE THREAD

**Issue by J0x90** [pgadmin-org/pgadmin4#7266]
**_Please note that security bugs or issues should be reported to security@pgadmin.org._**

**Describe the bug**

When attempting to run pgadmin4 with python (https://www.pgadmin.org/download/pgadmin-4-python/), I have observed that when entering the last command in the instructions, I get an error.
Command:
`pgadmin4`
Error:
`FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/pgadmin/pgadmin4.db' -> '/var/lib/pgadmin/pgadmin4.db.20240309203301'`

Looking through the source code, it appears that when the sqlite db does not exist, the program flow eventually ends up executing
`flask_migrate.upgrade(migration_folder)` within db_upgrade.py

For example, below we see upgrade_db() is entered because this is not cli_mode, and the SQLITE_PATH doesn't exist.
```
    def run_migration_for_sqlite():
        with app.app_context():
            # Run migration for the first time i.e. create database
            # If version not available, user must have aborted. Tables are not
            # created and so its an empty db
            if not os.path.exists(SQLITE_PATH) or get_version() == -1:
                # If running in cli mode then don't try to upgrade, just raise
                # the exception
                if not cli_mode:
                    upgrade_db()
                else:
                    if not os.path.exists(SQLITE_PATH):
                        raise FileNotFoundError(
                            'SQLite database file "' + SQLITE_PATH +
                            '" does not exists.')
                    raise RuntimeError(
                        'The configuration database file is not valid.')
```

Next we see db_upgrade(app) is entered.
```
    def upgrade_db():
        """
        Execute the migrations.
        """
        try:
            db_upgrade(app)
            os.environ['CORRUPTED_DB_BACKUP_FILE'] = ''
        except Exception:
            app.logger.error('Database migration failed')
            app.logger.error(traceback.format_exc())
            backup_db_file()
```

Finally, we see flask_migrate
```
def db_upgrade(app):
    from pgadmin.utils import u_encode, fs_encoding
    with app.app_context():
        migration_folder = os.path.join(
            os.path.dirname(os.path.realpath(u_encode(__file__, fs_encoding))),
            os.pardir, os.pardir,
            'migrations'
        )
        flask_migrate.upgrade(migration_folder)
```

Are yall sure that flask_migrate.upgrade is the right command to be ran for a first run of pgadmin4. Aka, will this create and initialize the db, because it appears that it is throwing an error when the db doesn't exist initially.

**To Reproduce**

1. Open terminal on Debian12
2. Enter the following commands
```
sudo adduser pgadmin4
sudo usermod -aG sudo pgadmin4
su - pgadmin4
sudo apt install python3.11-venv -y
sudo mkdir /var/lib/pgadmin
sudo mkdir /var/log/pgadmin
sudo chown $USER /var/lib/pgadmin
sudo chown $USER /var/log/pgadmin
python3 -m venv pgadmin4
source pgadmin4/bin/activate
pip install pgadmin4
pgadmin4
```
4. See the error
<img width="740" alt="out-1" src="https://github.com/pgadmin-org/pgadmin4/assets/12524177/8d95e6c6-1c51-4b1d-88f7-c5fe96b5116a">

**Expected behavior**

Successful installation.

**Error message**

If possible, include the full, unedited text of any error messages displayed.
```
FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/pgadmin/pgadmin4.db' -> '/var/lib/pgadmin/pgadmin4.db.20240309203909'
```

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Debian 12
 - Version: Linux debian-s-2vcpu-4gb-120gb-intel-sfo3-01 6.1.0-9-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.27-1 (2023-05-08) x86_64 GNU/Linux
 - Mode: Server
 - Browser (if running in server mode): NA
 - Package type: Python

**Additional context**

**Comment by khushboovashi** [comment]
@J0x90,
pgAdmin, first try to create the db file if it does not exist. You missed `create_app_data_directory(config)` in pgadmin/__init__.py file @ line no 252. There should be some permission issue on your end, so the db file wasn't created for you. 
Please provide the entire log file here.

**Comment by ShaheedHaque** [comment]
As per my thread at https://www.postgresql.org/message-id/CAHAc2je_TBjuZ86nBkAjjAmjgFmAhb03kLNaROSM7nxRoor1Zw%40mail.gmail.com, this problem occurs newly in flask-migrate 4.0.6. It looks as though the changes here:

https://github.com/miguelgrinberg/Flask-Migrate/commit/6f3f889c36030134f87dc1db327c2385d873a4d6

introduce some new behaviour which I guess pgadmin4 does not yet support.

**Comment by adityatoshniwal** [comment]
@J0x90 @ShaheedHaque Flask-Migrate has fixed the issue and released a new version - 4.0.7. Please upgrade your venv with latest Flask-Migrate

**Comment by deba-glitched** [comment]
flask-migrate 4.0.7 is not resolving the issue of the file not found error for the pgadmin4.db sqlite file
